pyensign/ensign.py

```diff
@@ -156,23 +152,34 @@ async def publish(
         if len(events) == 0:
             raise ValueError("no events provided")
 
-        # Resolve the topic ID from the name or ID string
         if ensure_exists:
+            # Get or create the topic if it doesn't exist
             id = ULID.from_str(await self.ensure_topic_exists(topic))
+            topic = Topic(id=id, name=topic)
         else:
-            id = self._resolve_topic(topic)
+            try:
+                # Try to resolve the topic ID from the string
+                topic = Topic(id=self.topics.resolve(topic))
+            except CacheMissError:
+                # Assume the topic string is a name
+                topic = Topic(name=topic)
```

Comment:
I think it would be helpful to add a comment here that at this point there is no topic_id defined in the Topic and that it will be updated when the stream is opened.